wire-ios-data-model/Source/Notifications/ObjectObserverTokens/Helpers/ClosureBasedUserChangeObserver.swift

```diff
@@ -0,0 +1,31 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+/// Allows for handling user change events via closure.
+public final class ClosureBasedUserChangeObserver: UserObserver {
```

Comment:
**[suggestion(blocking)]:** This could confirm to `ZMUserObserver` instead of the concrete class `UserObserver`.

Reply:
Or could have a nil property closure on existing `UserObserver` instead of a separate class doing the same thing. Consider this like the `URLSession` API where you could use closure way or async way
`var userDidChange:((UserChangeInfo) -> Void)?` add assign it where you need to observe it.

Reply:
Okay I deleted this class.